core/src/authentication.rs

```diff
@@ -42,6 +42,14 @@ impl Credentials {
         }
     }
 
+    pub fn with_access_token(username: impl Into<String>, token: impl Into<String>) -> Credentials {
```

Comment:
See https://github.com/librespot-org/librespot/pull/1098#issuecomment-1383155315
I think it would be good to not enforce the username here, if possible.

Reply:
Sure, I'll align to the work in https://github.com/librespot-org/librespot/pull/1098, which, I think, will add that part.